### PR TITLE
Added partial request printing functions

### PR DESCRIPTION
**Pull request by OinkIguana**
Was having trouble sending GET requests with this properly as the only "query only" string was pretty printed, so I added some functions to break up that one for writing POST requests. Changed the output format a bit too, so it would be a bit more readable/standard looking. Seems to be working from my app, and fixed up the tests that had to change for this. 

Not sure why there are so many commits listed. Seems like a lot of the ones from the last PR.

### CODE REVIEW

**Comment by VerachadW** [comment]
IMHO, I think it would be better to create an enum to represent print mode instead of using two combinations of 2 boolean flags. For example,
```kotlin
  enum class PrintMode {
    NORMAL,
    ONELINE,
    PRETTY;
  }
```
and in `requestQueryString()` would be 
```kotlin
   fun requestQueryString() = document.operation.print(PrintMode.ONELINE, 0)
```
Would it be better?

**Comment by OinkIguana** [comment]
Yeah that sounds like a better idea. I'll figure out a way to make it work. Maybe `NORMAL`, `PRETTY` and `JSON`?

**Comment by OinkIguana** [comment]
Alright it is changed. Rewrote the tests a lot too, so that there would be less stuff that gets duplicated. Doubt there would be another print format but if there is it should be easier to write tests for now.

File: README.md [hunk 1]
```diff
@@ -250,6 +250,12 @@ First, let's see what Kraph provides for you.
      * }
      */
     ```
+-   `requestQueryString()`, `requestVariableString()` and `requestOperationName()`
+    provide more fine grained access to the components of the full request string,
+    which are sometimes necessary depending on your HTTP request builder and
+    GraphQL server setup. They provide the values for the `query`, `variables`,
+    and `operationName` parameters, respectively, and so are good for creating
+    GET requests.
```
Comment (VerachadW) [inline README.md]: Please add a note here that `requestVariableString()` always return null for now. We will implement this after we support the variable.

**Pull request merged**